st2client/st2client/config_parser.py

```diff
@@ -79,6 +79,10 @@
         'password': {
             'type': 'string',
             'default': None
+        },
+        'apikey': {
```

Comment:
Do we even want to support API key in the config?

Config is mostly meant to be used by users and API keys are mostly meant to be used for integrating with 3rd party services so I would also be OK with not supporting API key in the CLI config.